libraries/c_sdk/standard/https/src/iot_https_client.c

```diff
@@ -2667,15 +2636,11 @@ IotHttpsReturnCode_t IotHttpsClient_Disconnect( IotHttpsConnectionHandle_t connH
         }
     }
 
-    IotMutex_Unlock( &( connHandle->respQMutex ) );
-
     /* Remove all pending requests. If this routine is called from the application context and there is a
      * network receive callback in process, this routine will wait in _networkDestroy until that routine returns.
      * If this is routine is called from the network receive callback context, then the destroy happens after the
      * network receive callback context returns. */
-    IotMutex_Lock( &( connHandle->reqQMutex ) );
     IotDeQueue_RemoveAll( &( connHandle->reqQ ), NULL, 0 );
-    IotMutex_Unlock( &( connHandle->reqQMutex ) );
 
     /* Do not attempt to destroy an already destroyed connection. This can happen when the user calls this function and
      * IOT_HTTPS_BUSY is returned. */
```

Comment:
Is there another routine that can be used to destroy the connection handle?

Reply:
Automatic disconnect happens on parsing and network errors during the taskpool job that sends the request and the network receive task that receives the response.